src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/Formatting/CSharpFormattingPass.cs

```diff
@@ -72,23 +73,17 @@ public async override Task<FormattingResult> ExecuteAsync(FormattingContext cont
 
             cancellationToken.ThrowIfCancellationRequested();
 
-            // Now, for each affected line in the edited version of the document, remove x amount of spaces
-            // at the front to account for extra indentation applied by the C# formatter.
-            // This should be based on context.
-            // For instance, lines inside @code/@functions block should be reduced one level
-            // and lines inside @{} should be reduced by two levels.
-            var indentationChanges = AdjustCSharpIndentation(changedContext, startLine: 0, endLine: changedText.Lines.Count - 1);
+            // We make an optimistic attempt at fixing corner cases.
+            changedText = CleanupDocument(changedContext);
```

Comment:
Because cleanup takes care of the remapping edge case which changes the line count.
```razor
@code {public class Foo { }
}
```
becomes,
```razor
@code {
    public class Foo { }
}
```
as part of `CleanupDocument`.

AdjustIndentation has to happen after because it looks at each individual lines and computes indentation.